### PR TITLE
Moving to a third party tool for diff

### PR DESCRIPTION
**Pull request by prateek2408**
Currently TFV test logs are hard to parse because the diff is output as a giant object. It would be nice to reduce the noice here and assist the developer in pinpointing the exact issues.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement

```

### CODE REVIEW

**Review by melinath — CHANGES_REQUESTED** [review]
Thanks! I have a few change requests. Also, could you provide a sample of what the failure output looks like after these changes?

It looks like the tests are currently failing for unrelated reasons.

File: mmv1/third_party/validator/tests/source/utils_test.go [hunk 1]
```diff
@@ -225,3 +227,20 @@ func getTestPrefix() string {
...
+	if err != nil {
+		panic(err)
+	}
+	changes, _ := d.Diff(actualAssets, expectedAssets)
+	if len(changes) == 0 {
+		return
+	}
+	t.Log("[Error] The following changes have occurred")
+	for _, i := range changes {
+		t.Log("[Error] The folllowing change has occoured")
```
Comment (melinath) [inline mmv1/third_party/validator/tests/source/utils_test.go:242]: This line is now a duplicate

```suggestion
```

**Review by melinath — CHANGES_REQUESTED** [review]
It looks like you will need to update calls to `getDiff` to instead call `assertAssetsMatch` - I think that will also mean that you can remove the import of github.com/stretchr/testify/assert for those files.

**Review by melinath — CHANGES_REQUESTED** [review]
Could you rebase this off the `main` branch and also set up a PR for the downstream?

**Comment by prateek2408** [comment]
> It looks like you will need to update calls to `getDiff` to instead call `assertAssetsMatch` - I think that will also mean that you can remove the import of github.com/stretchr/testify/assert for those files.

Rebase of branch diff_tool to main completed. Lets see if the test cases pass now.

**Comment by prateek2408** [comment]
I looked at the r3labs/diff package which we use and it seems there in terms of **type** there can only be 3 values "create, delete, update". So I refactored the logic to have a switch case instead and display different kind of log for each **type** 

So now If I take the previous example then this(Old) output is not changed to (New)-

Old-
```
            utils_test.go:238: [Error] The following changes have occurred
            utils_test.go:240:         Element [0 Ancestors 0] has been created from `<nil>` to  `projects/my-project-12345`
            utils_test.go:240:         Element [0 Ancestors 1] has been created from `<nil>` to  `folders/12345`
            utils_test.go:240:         Element [0 Ancestors 2] has been created from `<nil>` to  `organizations/67890`
```

New-
```
            utils_test.go:238: [Error] Differences Summary (+++ added,  -+ updated, --- deleted):
            utils_test.go:240:         Object [0 Ancestors 0] 
            utils_test.go:243:         +++ projects/my-project-12345
            utils_test.go:240:         Object [0 Ancestors 1] 
            utils_test.go:243:         +++ folders/12345
            utils_test.go:240:         Object [0 Ancestors 2] 
            utils_test.go:243:         +++ organizations/67890
```

Since we don't have the Name coming to the function so that is not getting displayed. Does this make sense ?
